### PR TITLE
Add return to removeMethodTemplate

### PR DESCRIPTION
**Pull request by aivus**
I think we should return result of remove operation

### CODE REVIEW

**Comment by Ocramius** [comment]
Why would that be required?

**Comment by aivus** [comment]
`ArrayCollection` return result of remove operation
https://github.com/doctrine/collections/blob/master/lib/Doctrine/Common/Collections/ArrayCollection.php#L119

I want to know whether the item was deleted.

**Comment by Ocramius** [comment]
@aivus can you then also update the docblock as per the [`Collection` interface](https://github.com/doctrine/collections/blob/6c1e4eef75f310ea1b3e30945e9f06e652128b8a/lib/Doctrine/Common/Collections/Collection.php#L98)?

**Comment by aivus** [comment]
Add `return` to docblock

**Comment by aivus** [comment]
@Ocramius should I do anything else?

File: lib/Doctrine/ORM/Tools/EntityGenerator.php [hunk 1]
```diff
@@ -277,10 +277,12 @@ public function <methodName>(<methodTypeHint>$<variableName>)
  * <description>
  *
  * @param <variableType> $<variableName>
+ *
+ * @return boolean TRUE if this collection contained the specified element, FALSE otherwise.
```
Comment (deeky666) [inline lib/Doctrine/ORM/Tools/EntityGenerator.php]: From a PSR-5 compat perspective, this should be `bool` not `boolean`.

Comment (aivus) [inline lib/Doctrine/ORM/Tools/EntityGenerator.php]: I get it from https://github.com/doctrine/collections/blob/6c1e4eef75f310ea1b3e30945e9f06e652128b8a/lib/Doctrine/Common/Collections/Collection.php#L98

Also PSR-5 is draft. I'm not sure about change `boolean` to `bool`. Is it required?

Comment (deeky666) [inline lib/Doctrine/ORM/Tools/EntityGenerator.php]: I don't care tbh, was just nitpicking because there was a PR about this kinda thing recently...

**Comment by deeky666** [comment]
Might be nice to have a regression test for this [here](https://github.com/doctrine/doctrine2/blob/master/tests/Doctrine/Tests/ORM/Tools/EntityGeneratorTest.php)

**Comment by aivus** [comment]
PR updated, tests added.

**Comment by aivus** [comment]
@Ocramius PR is ready. Please, review when you can.

**Comment by Ocramius** [comment]
@aivus thanks!

**Pull request merged**